app/models/mixins/authentication_mixin.rb

```diff
@@ -191,9 +191,31 @@ def authentication_type(type)
     end
   end
 
+  MAX_ATTEMPTS = 6
+  def authentication_check_retry_deliver_on(attempt)
+    # Existing callers who pass no attempt will have no delay.
+    case attempt
+    when nil, 0, ->(a) { a >= MAX_ATTEMPTS }
+      nil
+    else
+      Time.now.utc + exponential_delay(attempt - 1).minutes
+    end
+  end
+
+  def exponential_delay(attempt)
+    2**attempt
+  end
+
   def authentication_check_types_queue(*args)
-    options = args.extract_options!
+    method_options = args.extract_options!
     types = args.first
+    force = method_options.delete(:force) { false }
```

Comment:
Don't force the queue message by default, only the "retry" scenario should put a message without checking if the same auth check attempt/ deliver_on is there already.